examples/scripts/script_reslc.py

```diff
@@ -98,39 +98,79 @@ def get_free_port():
+    ifgs = sarxarray.from_binary(f_ifgs, shape, dtype=dtype_slc_ifg, chunks=reading_chunks) # Load ifgs
+    ifgs_h2ph = sarxarray.from_binary(f_h2phs, 
+                                      shape, 
+                                      vlabel='h2ph', 
+                                      dtype=dtype_lam_phi, 
+                                      chunks=reading_chunks) # Load h2phs
+    ifgs = ifgs.assign({"h2ph": ifgs_h2ph['h2ph'].astype(np.float16)}) # Add h2ph to ifgs
+    ifgs['time'] = [datetime.strptime(file.parts[-2], '%Y%m%d') for file in f_ifgs] # Add time coords to ifgs
+
+    # Lazy loading mother SLC 
+    mother = sarxarray.from_binary([f_mother_slc], shape, dtype=dtype_slc_ifg, chunks=reading_chunks) # Load mother SLC
+    # Construct a dummy h2ph with zeros and assign to mother
+    mother_h2ph = mother['amplitude'].copy()
+    mother_h2ph.data = da.zeros(mother_h2ph.shape, dtype=np.float16)
+    mother = mother.assign({"h2ph": mother_h2ph})
+    # Add time coords to mother
+    mother['time'] = [datetime.strptime(f_mother_slc.parts[-2], '%Y%m%d')]
 
     # Generate reconstructed SLCs
     slc_recon = ifg_to_slc(mother, ifgs)
 
     # Extract real and image part. remove other fields. convert to float16
+    # This applies to both slc_recon and mother
     slc_recon_output = slc_recon.copy()
+    slc_mother = mother.copy()
     slc_recon_output = slc_recon_output.assign(
         {
             "real": slc_recon_output["complex"].real.astype(np.float16),
             "imag": slc_recon_output["complex"].imag.astype(np.float16),
         }
     )
+    slc_mother = slc_mother.assign(
+        {
+            "real": mother['complex'].real.astype(np.float16),
+            "imag": mother['complex'].imag.astype(np.float16),
+        }
+    )
     slc_recon_output = slc_recon_output.drop_vars(["complex", "amplitude", "phase"])
+    slc_mother = slc_mother.drop_vars(['complex', 'amplitude', 'phase'])
+    
+    # Add mother SLC to the output
+    slcs_output = xr.concat([slc_recon_output, slc_mother], dim="time").sortby("time")
+
+    # Add geo-ref coords
+    lon = sarxarray.from_binary([f_lam], 
+                                shape, 
+                                vlabel='lon', 
+                                dtype=dtype_lam_phi, 
+                                chunks=reading_chunks).isel(time=0)['lon']
+    lat = sarxarray.from_binary([f_phi], 
+                                shape, 
+                                vlabel='lat', 
+                                dtype=dtype_lam_phi, 
+                                chunks=reading_chunks).isel(time=0)['lat']
+    slcs_output = slcs_output.assign({"lon": lon, "lat": lat})
 
     # Rechunk and write as zarr
-    slc_recon_output = slc_recon_output.chunk({"azimuth": writing_chunks[0], "range": writing_chunks[1]})
+    slcs_output = slcs_output.chunk(writing_chunks)
     if overwrite_zarr:
         slc_recon.to_zarr("nl_veenweiden_s1_dsc_t037.zarr", mode="w")
```

Comment:
Can we make this filename a parameter at the top of the script? So we give define all setitngs at the beginning.

Reply:
moved the output path to the file start variable `path_output_slcs`